### PR TITLE
Add a deployment pipeline

### PR DESCRIPTION
**Pull request by IAlibay**

### CODE REVIEW

**Comment by IAlibay** [comment]
This should give us wheels for the upcoming release.

@hmacdope @orbeckst I don't have keys for PYPI & TESTPYPI so I can't upload the right token - I might just do the OpenID connect migration here,

**Comment by IAlibay** [comment]
Tests passed fine - this is just waiting on setting up the trusted publisher.

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,145 @@
+name: Build and upload to PyPI
+
+on:
+  push:
+    tags:
+      - "*"
```
Comment (orbeckst) [inline .github/workflows/deploy.yaml:6]: You don't want to filter on specific version tags? Come to think that's fine given that we use these for testpypi.

**Review by orbeckst — APPROVED** [review]
nice!

File: .github/workflows/deploy.yaml (hunk 1 above)
Comment (IAlibay) [reply .github/workflows/deploy.yaml:6]: I do it in the core repo but I don't particularly feel like enforcing my tag habits on other, happy to change this to whichever pattern the pytng maintainers enjoy though.

Comment (orbeckst) [reply .github/workflows/deploy.yaml:6]: That's fair.

Reflecting a bit more, I'd also advocate for starting out less restrictive and tightening up if problems arise.

**Comment by IAlibay** [comment]
@orbeckst are you ok setting the trusted publisher details before merging? 

details should all be straightforward, workflow is `deploy.yaml`, environment is `deploy`

**Comment by IAlibay** [comment]
I can't block since I'm the author, but please hold merge for a second! There's an extra bit to updating the deployment file which I'm only finding out now.

File: .github/workflows/deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        with:
+          path: dist/*.tar.gz
+          retention-days: 7
+
+  upload_testpypi:
+    if: |
+      github.repository == 'MDAnalysis/pytng' &&
+      (github.event_name == 'push' && startsWith(github.ref, 'refs/tags/'))
+    name: testpypi upload
+    environment:
```
Comment (IAlibay) [inline .github/workflows/deploy.yaml:83]: Ok done - this was missing.

I also removed the post upload tests because pytng doesn't package test files.

**Comment by IAlibay** [comment]
Ok anticipated trusted publisher has been added on testpypi and the trusted publisher has been set properly on pypi.

**Pull request merged**